### PR TITLE
Expose objectId for mam support

### PR DESCRIPTION
**Pull request by Veena11**
## Proposed changes

Expose mdmId and objectId via extraDeviceInfo for MAM SDK , as mdmId and objectId wont be available via CP after JIT is rolled out. The new extraDeviceInfo dict  will be exposed via MSAL and one Auth apis

## Type of change

- [ x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.h [hunk 1]
```diff
@@ -50,8 +50,8 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic) MSIDWorkPlaceJoinStatus wpjStatus;
 @property (nonatomic, nullable) NSString *brokerVersion;
 @property (nonatomic) NSDictionary *additionalExtensionData;
-// New property to return MDM Id to 1P and managed apps
-@property (nonatomic, nullable) NSString *mdmId;
+// New property to return additional device Info
+@property (nonatomic) NSMutableDictionary *extraDeviceInfo;
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDDeviceInfo.h:54]: Should be non-mutable dictionary to follow best practices

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m [hunk 2]
```diff
@@ -63,7 +76,7 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(__unused NSEr
         
         NSData *jsonData = [json[MSID_ADDITIONAL_EXTENSION_DATA_KEY] dataUsingEncoding:NSUTF8StringEncoding];
         _additionalExtensionData = [NSJSONSerialization msidNormalizedDictionaryFromJsonData:jsonData error:nil];
-        _mdmId = [json msidStringObjectForKey:MSID_BROKER_MDM_ID_KEY];
+        _extraDeviceInfo = [json msidObjectForKey:MSID_EXTRA_DEVICE_INFO_KEY ofClass:[NSDictionary class]];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:79]: Looks like you're initializing this with a non-mutable dictionary

Comment (kaisong1990) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:79]: I think it is ok to assign a reference to a non-mutable dictionary, as long as we don't change anything inside the dictionary

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.h (hunk 1 above)
Comment (Veena11) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.h]: Addressed

File: IdentityCore/src/MSIDBrokerConstants.m [hunk 3]
```diff
@@ -76,3 +76,5 @@
 NSString *const MSID_ADDITIONAL_EXTENSION_DATA_KEY = @"additional_extension_data";
 NSString *const MSID_SSO_NONCE_QUERY_PARAM_KEY = @"sso_nonce";
 NSString *const MSID_BROKER_MDM_ID_KEY          = @"mdm_id";
+NSString *const MSID_ENROLLED_USER_OBJECT_ID_KEY          = @"object_id";
```
[the remark this answers is not in the document]
Comment (Veena11) [reply IdentityCore/src/MSIDBrokerConstants.m]: done

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m (hunk 2 above)
Comment (Veena11) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:67]: changed to non mutable dict

**Review by oldalton — APPROVED** [review]

**Pull request merged**